### DISCUSSION TITLE
PostProcessing/Shaders broken

### DISCUSSION THREAD

**Discussion by blubblub-web** [AssetRipper/AssetRipper#643] [Q&A]
Hi,
I unpacked a game made with Unity 5.6.6 and everything works fine except for the post processing, which let almost everything rendered as black, with a lot of these errors in the console 
"Invalid pass number (4) for Graphics.Blit (Material "(Unknown material)" with 4 passes)".

I saw that turning off the post processing in the cam, it fixes partially the issue, but this doesn't look to me like a real solution.
![Unity 5 6 6f2 Personal (64bit) - titlescreen unity - ExportedProject - PC, Mac   Linux Standalone_ _DX11_ 17_12_2022 01_17_17](https://user-images.githubusercontent.com/63713882/208212446-62559a0f-b057-4562-b1c2-18de8ad59297.png)

**Comment by ds5678** [comment]
@nesrak1 would be able to answer better, but I think shader decompilation only works well for a subset of shader types.

**Comment by nesrak1** [comment]
I haven't seen this one. Does Unity show you which line this happens on? If so, can I get a bit of context what line it is? It could just be the metadata (not my code) and not related to the decompilation code at all.

**Reply by nesrak1** [reply]
Ok so a little more research into this shows it's a very generic error and just means that _something_ went wrong but not what went wrong. I guess I should ask if there are any other errors other than the Invalid pass number one, or if that's all.

**Reply by blubblub-web** [reply]
No no, sadly that's it. I did some research too, to try to fix it with no results tho...
As I said, turning off the post processing attachments from the cam solves partially the issue and, I also tried changing all the material's shaders to "Standard" without results. I even tried ripping again the shaders with Yalm (instead Dummy which I used the 1st time) and still same error...

**Reply by nesrak1** [reply]
Ah, so this was all in Dummy mode and not Decompile or Disassembly mode? I was thinking it could be that, but you never said which mode you used in the original post. That changes a whole lot then, the dummy shaders hardly have any code in them.

**Reply by blubblub-web** [reply]
You're right. My bad then. What about the other error I posted earlier?

**Reply by nesrak1** [reply]
No idea. I haven't seen that either, but that's a script error (from C# code) rather than a shader compile error. I would expect a shader compile error somewhere in that list then.

**Comment by blubblub-web** [comment]
Edit. I tried re-exporting the whole project but, this time, the shaders directly as Yalm. 
And I didn't get that issue. Instead, cameras doesn't work at all. I mean they're disabled and if I run the game they keep being disabled, and I get these errors in the console, instead.

![Unity 5 6 6f2 Personal (64bit) - titlescreen unity - ExportedProject - PC, Mac   Linux Standalone_ _DX11_ 17_12_2022 15_37_13](https://user-images.githubusercontent.com/63713882/208247423-eee65127-b6f8-4e4b-b2b2-6d39f6a59aec.png)

**Comment by csu01** [comment]
这是来自蔡文杰邮箱的自动回复邮件。
 
您好，蔡文杰已收到收到您的来信，我会尽快给你回复。

**Reply by ds5678** [reply]
这是一封垃圾邮件。

**Comment by HeDevedUp** [comment]
Link is not working

**Reply by ds5678** [reply]
It was a spam message.

**Comment by lolyparty** [comment]
Spam email?

**Reply by ds5678** [reply]
Yes, some bot wanted attention.
